src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -119,6 +126,7 @@ object ScalafixPlugin extends AutoPlugin {
     scalafixCaching := false,
     scalafixResolvers := ScalafixCoursier.defaultResolvers,
     scalafixDependencies := Nil,
+    scalafixScalaVersion := BuildInfo.scala212, // will follow scalafixVersion once community rules are cross-built
```

Comment:
I wonder if it's not worth making this a `TaskKey` (or `scalafixDependencies` a `SettingKey`?), to be able to support https://github.com/scalacenter/scalafix/issues/998 automatically when there is no external rule with something like:
```suggestion
    scalafixScalaVersion := {
      if (scalafixDependencies.value.isEmpty) scalaVersion.value
      else BuildInfo.scala212
    }
```

This can break invocations with `dependency:`, but maybe that's OK as long as the documentation recommends to force `scalafixScalaVersion := BuildInfo.scala212` until 1.x?

Reply:
> Currently, all rules need to be compiled in 2.12 to be used. Right?

Considering the [git history here](https://github.com/scalacenter/sbt-scalafix/commit/a8937c5398abe791334bf82f6860949b486d1d6f#diff-61ff4bfeba35dc5df102214ffbf71224R16) which shows that sbt-scalafix has always been loading rules (built-in or external) in 2.12 and the repos references in the user documentation (https://github.com/olafurpg/example-scalafix-rule & https://github.com/olafurpg/named-literal-arguments), I think it's fair to assume that, yes.

Reply:
>  I guess we hope that most rules will be cross-compiled soon.

My suggestion in the comment above was more for the transition period: we can leverage the fact that `scalafix-rules` IS cross-built to 2.11/2.13, so picking it up fixes https://github.com/scalacenter/scalafix/issues/998. However, it's safe to do so if there are no external rules as part of the same invocation. If there are, we can't be sure they are cross-compiled, at least not in the near future (0.9.x).

Reply:
The scala version must match exactly in scalafix-cli_$SCALA_VERSION. I don't think we can use `scalaVersion.value` from the project because that would break if users have a mismatch on the patch version (2.13.1 vs 2.13.2).

There are rare cases where using scalafix 2.13.2 with for example scala version 2.13.3 could cause runtime exceptions. I think we shouldn't worry about them since the cost of building against each exact version is too high when compared to the cost of hitting these rare runtime errors (which can often be worked around on the user side by upgrading to a more recent Scala version).

Reply:
> I don't think we can use scalaVersion.value from the project because that would break if users have a mismatch on the patch version (2.13.1 vs 2.13.2).

Good point, the scripted doesn't exercise this since we use `scalafix.sbt.BuildInfo.scala*` as `scalaVersion`, I will update that.

Considering we add a `projectVersionToScalafixVersion(String: scalaVersion): String` which takes care of looking up the correct patch version based on the binary version the project is built with, what do you think of my proposal

in 0.9.x
```suggestion
    scalafixScalaVersion := {
      if (scalafixDependencies.value.isEmpty) projectVersionToScalafixVersion(scalaVersion.value)
      else BuildInfo.scala212
    }
```

in 1.x
```suggestion
    scalafixScalaVersion := projectVersionToScalafixVersion(scalaVersion.value)
```